firedrake/slate/slate.py

```diff
@@ -553,10 +576,20 @@ def arguments(self):
         """Returns a tuple of arguments associated with the tensor."""
         return self._split_arguments
 
+    @cached_property
+    def form(self):
+        tensor, = self.operands
+        if tensor.rank > 1:
+            return ExtractSubBlock().split(tensor.form, self._indices)
+        elif tensor.rank <= 1:
+            return tuple(tensor._function.split()[i] for j in self._indices for i in j)
```

Comment:
Style: I kind of prefer `for i in itertools.chain(*self._indices)`

Reply:
Yes, I like that better, too.

Reply:
So this property would be the "create_subtensor" thing.